### ISSUE TITLE
Verification Request

### ISSUE THREAD

**Issue by prmsregmi** [khalti/khalti-sdk-web#25]
Khalti server is verifying my request via localhost. I am using AJAX to make asynchronous call to my server. But when I try with HTTP only web-hosting service of Azure, the verification request does not move forward. Am I supposed to use HTTPS or there is some problem in my code. I tested, there is no problem with Ajax Call in the production environment.

**Comment by chapagainmanoj** [comment]
can you provide more details, what error you get? Did you call verification api with ajax?

**Comment by prmsregmi** [comment]
The request did not move forward.

```
onSuccess (payload) {
                    // hit merchant api for initiating verfication
                $.ajax({
                    url: '/khalti_payment/',
                    data: {
                    'success': true,
                    'order_id': prod_id,
                    'amount': amt,
                    'token': payload.token
                    },
                    dataType: 'json'
                });
```

I called the verification API with Ajax. The verification API was in Django (Python)

```
def khalti_payment(request):
    if request.GET.get('success'):
        order_id = request.GET.get('order_id')
        amount = request.GET.get('amount')
        code = request.GET.get('code')
        token = request.GET.get('token')
       
       #start of verification
        import requests
        url = "https://khalti.com/api/v2/payment/verify/"
        payload = {
            "token": token,
            "amount": amount,
        }
        headers = {
        "Authorization": "Key test_secret_key_5fd0bd5e122e4bf0832e9c1f272b3316"
        }
        response = requests.post(url, payload, headers = headers)
        #end of verification

        if response.status_code==200:
                 #task I do when payment verified

        return HttpResponse(status=200)
}
```

The web app reached upto "#start of verification" part and not to "#end of verification" part. The same code worked in my localhost. So, I though, maybe there was some issue with the Khalti server.

**Comment by prmsregmi** [comment]
The request did not move forward.

```
onSuccess (payload) {
                    // hit merchant api for initiating verfication
                $.ajax({
                    url: '/khalti_payment/',
                    data: {
                    'success': true,
                    'order_id': prod_id,
                    'amount': amt,
                    'token': payload.token
                    },
                    dataType: 'json'
                });
```

I called the verification API with Ajax. The verification API was in Django (Python)

```
def khalti_payment(request):
    if request.GET.get('success'):
        order_id = request.GET.get('order_id')
        amount = request.GET.get('amount')
        code = request.GET.get('code')
        token = request.GET.get('token')
       
       #start of verification
        import requests
        url = "https://khalti.com/api/v2/payment/verify/"
        payload = {
            "token": token,
            "amount": amount,
        }
        headers = {
        "Authorization": "Key test_secret_key_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
        }
        response = requests.post(url, payload, headers = headers)
        #end of verification

        if response.status_code==200:
                 #task I do when payment verified

        return HttpResponse(status=200)
}
```

The web app reached upto "#start of verification" part and not to "#end of verification" part. The same code worked in my localhost. So, I though, maybe there was some issue with the Khalti server.

**Comment by prmsregmi** [comment]
Apparently, there was some issues with the code only. I overlooked the fact that requests is not included in python itself. Still, the microsoft server was slow to adapt the changes, thus the issue persisted even after I installed request. Now the issue is solved. Thanks for the fast reply. And, I really want to tell you that Khalti team has done a great job in this SDK and Verification system. Keep the good work.
